### DISCUSSION TITLE
How to Container.RegisterInitializer<TService>(Action<TService>) to use Jot.

### DISCUSSION THREAD

**Discussion by michalciolek** [DevTeam/Pure.DI#25] [Q&A]
I would like to RegisterInitializer (like [SimpleInjector](https://simpleinjector.org/ReferenceLibrary/html/M_SimpleInjector_Container_RegisterInitializer__1.htm)) using PureDI. It is possible?
I need Integrate Jot with my DI: https://github.com/anakic/Jot#ioc-integration (call `tracker.Track()` on all Form instances after ctor)

**Comment by NikolayPianikov** [comment]
Yes, it is possible using some implementation of [IFactory](https://github.com/DevTeam/Pure.DI/blob/ab2934e6c71c47be4b598cb7d6cc11cfa270f0e8/Pure.DI/Components/Contracts.cs#L719) or [generic IFactory](https://github.com/DevTeam/Pure.DI/blob/ab2934e6c71c47be4b598cb7d6cc11cfa270f0e8/Pure.DI/Components/Contracts.cs#L702). Please see [this sample](https://github.com/DevTeam/Pure.DI#custom-singleton-lifetime) or [this](https://github.com/DevTeam/Pure.DI#intercept-specific-types).

**Comment by michalciolek** [comment]
Yes, this is what I looking for.
Now I use nongeneric IFactory, with `[Include(".*View$")]` but I cannot filter some classes.
I have Forms and UserControls whose names end with `View`. How I can filter using base class instead of class name?
When I create a generic `IFactory<T> where T : Form`) I get `There is no implicit reference conversion from SomeUserControlView to Form`.

**Comment by NikolayPianikov** [comment]
The "Include" filter works at compile time, but at run time you can perform additional filtering on already filtered types:

```
var val = factory();
if (val is BaseType) { }
```

This will work quite efficiently if you reduce the number of types at compile time, type checking is fast.

**Reply by michalciolek** [reply]
It would be nice if V2 could filter by BaseType during complication :)
